src/main/java/org/opensearch/neuralsearch/processor/DocumentChunkingProcessor.java

```diff
@@ -0,0 +1,328 @@
+
+        for (Map.Entry<String, Object> algorithmEntry : algorithmMap.entrySet()) {
+            String algorithmKey = algorithmEntry.getKey();
+            Object algorithmValue = algorithmEntry.getValue();
+            if (!supportedChunkers.contains(algorithmKey)) {
+                throw new IllegalArgumentException(
+                    "Unable to create the processor as chunker algorithm ["
+                        + algorithmKey
+                        + "] is not supported. Supported chunkers types are ["
+                        + FIXED_LENGTH_ALGORITHM
+                        + ", "
+                        + DELIMITER_ALGORITHM
+                        + "]"
+                );
+            }
+            if (!(algorithmValue instanceof Map)) {
+                throw new IllegalArgumentException(
+                    "Unable to create the processor as [" + ALGORITHM_FIELD + "] cannot be cast to [" + Map.class.getName() + "]"
+                );
+            }
+            FieldChunker chunker = ChunkerFactory.create(algorithmKey, analysisRegistry);
+            chunker.validateParameters((Map<String, Object>) algorithmValue);
+            this.chunkerType = algorithmKey;
+            this.chunkerParameters = (Map<String, Object>) algorithmValue;
+            chunker.validateParameters(chunkerParameters);
+            if (((Map<String, Object>) algorithmValue).containsKey(MAX_CHUNK_LIMIT_FIELD)) {
+                int max_chunk_limit = ((Number) ((Map<String, Object>) algorithmValue).get(MAX_CHUNK_LIMIT_FIELD)).intValue();
+                if (max_chunk_limit <= 0) {
+                    throw new IllegalArgumentException("Parameter [" + MAX_CHUNK_LIMIT_FIELD + "] must be a positive integer");
+                }
+                this.max_chunk_limit = max_chunk_limit;
+            }
+        }
+    }
+
+    @SuppressWarnings("unchecked")
+    private boolean isListString(Object value) {
+        // an empty list is also List<String>
+        if (!(value instanceof List)) {
+            return false;
+        }
+        for (Object element : (List<Object>) value) {
+            if (!(element instanceof String)) {
+                return false;
+            }
+        }
+        return true;
+    }
+
+    private List<String> chunkString(String content) {
+        FieldChunker chunker = ChunkerFactory.create(chunkerType, analysisRegistry);
+        List<String> result = chunker.chunk(content, chunkerParameters);
+        current_chunk_count += result.size();
+        if (max_chunk_limit != DEFAULT_MAX_CHUNK_LIMIT && current_chunk_count > max_chunk_limit) {
+            throw new IllegalArgumentException(
+                "Unable to create the processor as the number of chunks ["
+                    + current_chunk_count
+                    + "] exceeds the maximum chunk limit ["
+                    + max_chunk_limit
+                    + "]"
+            );
+        }
+        return chunker.chunk(content, chunkerParameters);
```

Comment:
Return the result directly and don't need to chunk again.

Reply:
Already changed.